tools/cache_creator/units/cache_creator_tests.cpp

```diff
@@ -128,3 +129,82 @@ TEST_CASE("Full UUID roundtrip") {
   CHECK(cc::hexStringToUuid(hexStr, dumped));
   CHECK(dumped == uuid);
 }
+
+TEST_CASE("Get cache info from invalid PAL metadata blob") {
+  char badMetadata[10] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+
+  llvm::StringRef noteBlob(badMetadata, sizeof(badMetadata));
+  llvm::Expected<cc::ElfLlpcCacheInfo> cacheInfoOrErr = cc::getCacheInfoFromMetadataBlob(noteBlob);
+
+  llvm::Error err = cacheInfoOrErr.takeError();
+  CHECK(err.isA<llvm::StringError>());
+
+  llvm::consumeError(std::move(err));
+}
+
+TEST_CASE("Get cache info from valid PAL metadata blob") {
+  const char *sampleMetadata = "---\n"
+                               "amdpal.pipelines:\n"
+                               "  - .api:            Vulkan\n"
+                               "    .hardware_stages:\n"
+                               "      .cs:\n"
```

Comment:
I think we can drop unrelated metadata entries. Full pipeline metadata gets exercised in lit tests and we should use unit tests to make sure that the code can read it with only the necessary parts in place. This would give us confidence that the code will survive future unrelated changes to the metadata format.